toml/src/main/kotlin/org/rust/toml/CratesIoApi.kt

```diff
@@ -36,21 +34,36 @@ val CrateDescription.dependencyLine: String
 fun searchCrate(key: TomlKey): Collection<CrateDescription> {
     if (isUnitTestMode) return MOCK!!
 
-    val name = CompletionUtil.getOriginalElement(key)?.text ?: ""
+    // Stuff like [dependencies.'crate'] can contain single quotes which we need to remove or otherwise the crates API
+    // request would not work
+    val name = CompletionUtil.getOriginalElement(key)?.text?.removeSurrounding("'") ?: ""
     if (name.isEmpty()) return emptyList()
 
     val response = requestCratesIo<SearchResult>(key, "crates?page=1&per_page=20&q=$name&sort=") ?: return emptyList()
     return response.crates
 }
 
-fun getCrateLastVersion(key: TomlKey): String? {
-    if (isUnitTestMode) return MOCK!!.first().maxVersion
+data class CrateFullDescription(
+    val name: String,
+    @SerializedName("max_version")
+    val maxVersion: String,
+    val versions: List<CrateVersionDescription>
+)
+
+data class CrateVersionDescription(val id: Int, val num: String, val yanked: Boolean)
+
+private class CratesIoApiResponse(val crate: CrateDescription, val versions: List<CrateVersionDescription>)
+
+fun getCrateFullDescription(crateName: TomlKey): CrateFullDescription? =
+// Stuff like [dependencies.'crate'] can contain single quotes which we need to remove or otherwise the crates API
+    // request would not work
+    getCrateFullDescription(crateName, crateName.text.removeSurrounding("'"))
```

Comment:
Maybe extract this code to avoid duplication (if such simple removal really works)?